addons/website_mass_mailing/tests/test_snippets.py

```diff
@@ -16,4 +16,10 @@ def test_01_newsletter_popup(self):
         self.assertIn("hello@world.com", emails)
 
     def test_02_newsletter_block_edition(self):
+        admin_id = self.env.ref('base.user_admin').id
+        mailing_list = self.env['mailing.list'].search([('contact_ids', 'in', [admin_id])])
+        # Remove the user so that it can properly subscribe during the tour
+        mailing_list.write({
+            'contact_ids': [odoo.Command.delete(admin_id)]
```

Comment:
@detrouxdev since it's just linked to this test, wouldn't be better to just tweak the JS to use a newsletter to which the admin is not subscribed instead of using the `delete` command? Something like:
```diff
     {
         content: 'Wait for the list id to be set.',
         trigger: 'iframe .s_newsletter_block[data-list-id]:not([data-list-id="0"]) .s_newsletter_subscribe_form',
-        run: () => null, // it's a check
+    },
+    {
+        content: 'Click on the newsletter selector',
+        trigger: '[data-attribute-name="listId"] we-toggler',
+    },
+    {
+        content: 'Select a newsletter to which the user is not subscribed',
+        trigger: '[data-attribute-name="listId"] we-button:eq(1)',
     },
```
What do you think ?

Reply:
Thanks for checking it out! 

I'm not sure I understand the benefits of doing it in the JS side. From what you're doing, we're unsure of which mailing list will be selected, and it could be one from which the user is subscribed to anyway.

At least from the python side, we always ensure that the test env will be a valid one. The commands done during a test are rolled back at the end of that test. And that python test only runs that one tour, and that tour is only ran from python.

Reply:
> At least from the python side, we always ensure that the test env will be a valid one.

@detrouxdev Yes the Python solution is more generic but I don't think it should be implemented this way:

Here the `write(...Command.delete())` code will try to "remove" & "unlink" a `'mailing.contact'` record from the mailing list using the `id` you get from `'base.user_admin'`, which will target another contact while the admin is still subscribed to the mailing list... The strange thing is that, somehow, after the `delete` command, the `search()` & `name_search()`,... methods will always return a `'mailing.list'` recordset in reverse order :confused:, which makes the second mailing list (to which the admin is not subscribed) fetched first and used in the JS tour, and as a consequence, the test works correctly...

By reading the code on the `mass_mailing` > `'mailing.list'` model, I can see it has an [_order = 'create_date DESC'](https://github.com/odoo/odoo/blob/16.0/addons/mass_mailing/models/mailing_list.py#L14) right after [_order = 'name'](https://github.com/odoo/odoo/blob/16.0/addons/mass_mailing/models/mailing_list.py#L11) :thinking:. So, for example, if you simply force the `_order` to `'id'` in the model, you will see that the tour test will fail again...

I just checked this code using odoo `shell`, so maybe I missed something here... Can you check this and maybe just "unlink" the `'mailing.contact'` records with the same email as the admin... :+1:

Reply:
Indeed this first implementation was naive. I pushed an updated one which does what you asked!